spinalcordtoolbox/scripts/sct_resample.py

```diff
@@ -122,22 +125,22 @@ def main(argv: Sequence[str]):
         param.new_size = arguments.f
         param.new_size_type = 'factor'
         arg += 1
-    elif arguments.mm is not None:
+    if arguments.mm is not None:
         param.new_size = arguments.mm
         param.new_size_type = 'mm'
         arg += 1
-    elif arguments.vox is not None:
+    if arguments.vox is not None:
         param.new_size = arguments.vox
         param.new_size_type = 'vox'
         arg += 1
-    elif arguments.ref is not None:
+    if arguments.ref is not None:
         param.ref = arguments.ref
         arg += 1
-    else:
-        printv(parser.error('ERROR: you need to specify one of those three arguments : -f, -mm or -vox'))
 
-    if arg > 1:
-        printv(parser.error('ERROR: you need to specify ONLY one of those three arguments : -f, -mm or -vox'))
+    if arg == 0:
+        parser.error("You need to specify one of those three arguments: '-f', '-mm' or '-vox'.")
+    elif arg > 1:
+        parser.error("You need to specify ONLY one of those three arguments: '-f', '-mm' or '-vox'.")
```

Comment:
Looking into the source code, `-ref` gets passed to `resample_file()`, which then gets passed to `resample_nib()` via the `image_dest` parameter.

The docstring for `image_dest` says:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/54325c0dbc2bad5a72e4908f57c27550e762a76c/spinalcordtoolbox/resampling.py#L37-L38

Ignored? In that case, `-ref` should be treated as **mutually exclusive** with `-f`, `-mm`, and `-vox`, right?

Reply:
To address this, I would suggest to:

- [ ] Move the `-ref` option to the `resample_types` argument group, so that it's clear to users that they specify `-ref` _or_ one the other options.
- [ ] Update the `resample_types` argument group description, to say something like `METHOD TO SPECIFY NEW SIZE` and mention to choose one of 4 options.
- [ ] Update the `-ref` help string to clarify that the `-ref` image will be used for its voxel dimensions and affine.
    - Source: In the `nibabel` source code for `resample_from_to()`, it says:
    ```python
        """
        to_vox_map : image object or length 2 sequence
        If object, has attributes ``shape`` giving input voxel shape, and
        ``affine`` giving mapping of input voxels to output space
        """
    ```
- [ ] Update the error help strings to mention "one of four options" and include `-ref` in the list.

Reply:
Awesome catch! 👏🏻 

>  `-ref` should be treated as **mutually exclusive** with `-f`, `-mm`, and `-vox`, right?

I think so!

I moved the `-ref` option to the `resample_types` and addressed the points you suggested. See https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4081/commits/8dc5b5e95925702ed6bec078c8036172eb507bfa.